src/index.d.ts

```diff
@@ -0,0 +1,180 @@
+import { BufferGeometry, EdgesGeometry, Group, InstancedMesh, LineSegments, Mesh, MeshPhongMaterial, Plane, ShaderMaterial, Triangle, Vector3 } from 'three';
+
+export class Brush extends Mesh {
+
+  isBrush: boolean;
+  markUpdated(): void;
+  isDirty(): boolean;
+  prepareGeometry(): void;
+  disposeCacheData(): void;
+
+}
+
+export class TypedAttributeData {
+
+  groupAttributes: Object[];
+  groupCount: number;
+
+  getType( name: String ): string;
+  getTotalLength( name: String ): number;
+  getGroupSet( index?: number ): Object;
+  getGroupArray( name: string, index?: number ): Array; // return Array ???
+  initializeArray( name: string, type: string ): void;
+  clear(): void;
+  delete( key: string ): void;
+  reset(): void;
+
+}
+
+export enum OperationStrategy{} // Is the name correct?
+export const ADDITION: OperationStrategy;
+export const SUBTRACTION: OperationStrategy;
+export const DIFFERENCE: OperationStrategy;
+export const INTERSECTION: OperationStrategy;
+
+export class Evaluator {
+
+  triangleSplitter: TriangleSplitter;
+  attributeData: TypedAttributeData;
+  attributes: String[];
+  useGroups: boolean;
+  debug: OperationDebugData;
+
+  evaluate( a: Brush, b: Brush, operation: OperationStrategy, targetBrush?: Brush ): Brush;
+
+  evaluateHierarchy( root: Brush, target?: Brush ): Brush; // Root is Brush type??
+
+}
+
+export class Operation extends Brush {
+
+  isOperation: boolean;
+  // operation
+  markUpdated(): void;
+  isDirty(): boolean;
+  insertBefore( brush: Brush );
+  insertAfter( brush: Brush );
+
+}
+
+export class OperationGroup extends Group {
+
+  isOperatioinGroup: boolean;
+  markUpdated(): void;
+  isDirty(): boolean;
+
+}
+
+export class CullableTriangle extends Triangle {
+
+  initFrom( other: Triangle );
+  updateSide( plane: Plane, triangle: Triangle, coplanarIndex: number );
+
+}
+
+export class TrianglePool {
+
+  getTriangle(): Triangle;
+  clear(): void;
+  reset(): void;
+
+}
+
+export class TriangleSplitter {
+
+  trianglePool: TrianglePool;
+  triangles: Triangle[];
+  normal: Vector3;
+
+  initialize( tri: Triangle );
+  splitByTriangle( triangle: Triangle );
+  splitByPlane( plane: Plane, triangle: Triangle, coplanarIndex: number );
+  reset(): void;
+
+}
+
+export class HalfEdgeMap {
+
+  constructor( geometry?: BufferGeometry );
+  getSiblingTriangleIndex( triIndex: number, edgeIndex: number ): number;
+  getSiblingEdgeIndex( triIndex: number, edgeIndex: number ): number;
+  updateFrom( geomtry: BufferGeometry ): void;
+
+}
+
+export class GridMaterial extends MeshPhongMaterial {
+
+  enableGrid: boolean;
+
+  onBeforeCompile( shader: ShaderMaterial ): void;
+  customProgramCacheKey(): boolean;
```

Comment:
These two are defined on `MeshPhongMaterial` so there's no need to declare them on the types here.